MapboxCoreNavigation/RouteController.swift

```diff
@@ -134,6 +134,18 @@ public protocol RouteControllerDelegate: class {
     */
     @objc(routeController:didArriveAtWaypoint:)
     optional func routeController(_ routeController: RouteController, didArriveAt waypoint: Waypoint) -> Bool
+    
+    /**
+     Called when the route controller arrives at a waypoint.
+     
+     You can implement this method to allow the route controller to continue check and reroute the user if needed. By default, the user will not be rerouted when arriving at a waypoint.
+     
+     - parameter routeController: The route controller that has arrived at a waypoint.
+     - parameter waypoint: The waypoint that the controller has arrived at.
+     - returns: True to prevent the route controller from checking if the user should be rerouted.
+     */
+    @objc(routeController:shouldPreventReroutesWhenArrivingAtWaypoint:)
+    optional func routeController(_ routeController: RouteController, shouldPreventReroutesWhenArrivingAt waypoint: Waypoint) -> Bool
```

Comment:
Can anyone think of a use case where a developer would return false here? If so, maybe we do not actually need to implement a delegate. 

Stopping rerouting after arriving seems like a good feature for all.

Reply:
Ok, will keep.

Reply:
I think we should revisit in the process of implementing a sample ride-hailing driver app.